code/game/gamemodes/traitor/traitor.dm

```diff
@@ -56,15 +52,58 @@
 
 	if(!length(pre_traitors))
 		return FALSE
-	return TRUE
-
+	return LATE_HANDOUT
 
 /datum/game_mode/traitor/post_setup()
-	for(var/t in pre_traitors)
-		var/datum/mind/traitor = t
-		traitor.add_antag_datum(/datum/antagonist/traitor)
-	..()
+	. = ..()
 
+	for(var/datum/mind/traitor in pre_traitors)
+		var/datum/antagonist/traitor/traitor_datum = new(src)
+		traitor_datum.give_objectives = FALSE
+		traitor_datum.add_antag_objective(/datum/objective/traitor_wait)
+		traitor.add_antag_datum(traitor_datum)
+		traitors += src
```

Comment:
```suggestion
```
Think you meant `traitors += traitor`, but anyway, that already happens in `/datum/antagonist/traitor/add_owner_to_gamemode()`, so with this, there would be two duplicate mind refs for every traitor there is.

Reply:
Whoops, you're right, I'll remove it tomorrow